src/ModuleHomomorphism.jl

```diff
@@ -59,28 +59,15 @@ function kernel(f::Map(FPModuleHomomorphism))
    crels = rels(C)
    M = matrix(f)
    # put domain relations and M in a big matrix
-   # swap rows so we can get upper triangular wrt original data
    nr = nrows(M) + length(crels)
-   N = zero_matrix(R, nr, ncols(M))
-   for i = 1:nrows(M)
-      for j = 1:ncols(M)
-         N[nr - i + 1, j] = M[i, j]
-      end
-   end
-   for i = 1:length(crels)
-      for j = 1:ncols(M)
-         N[nr - i - nrows(M) + 1, j] = crels[i][1, j]
-      end
+   N = M
+   if length(crels) != 0
+      NN = reduce(vcat, crels)
+      N = vcat(N, NN)
    end
    # compute the kernel
-   num_gens, K = AbstractAlgebra._left_kernel(N)
-   # Construct generators of kernel submodule, reversing rows
-   # and columns so they're correct wrt to original data and
-   # in upper triangular form
-   V = Vector{elem_type(D)}(undef, num_gens)
-   for j = 1:num_gens
-      V[j] = D([K[num_gens - j + 1, nr - k + 1] for k = 1:nrows(M)])
-   end
+   K = AbstractAlgebra.kernel(N)
+   V = [D(K[j:j, 1:nrows(M)]) for j in 1:nrows(K)]
    return sub(D, V)
```

Comment:
I'll have a look, hopefully tomorrow. I do not really understand the comments either and would be very happy to get rid of this....

Reply:
I still don't understand Bill's reasoning. But as a side, should we do 
```V = [D(view(K[j:j, 1:nrows(M)])) for j in 1:nrows(K)]```
to avoid copying? Or does `sub` need an independent copy?
I prefer your version: it is readable

Reply:
I tried using `view`, but the problem is that it does not produce a `MatElem` for the AbstractAlgebra matrices but some special `View` type. And from that one cannot create a module element because they want `dense_matrix_type(...)`.
For the flint types it would work because there `view` again creates an element of type `FqMatrix` etc.